openedx/core/djangoapps/user_authn/views/tests/test_login.py

```diff
@@ -1179,29 +1179,18 @@ def test_invalid_credentials(self):
         })
         self.assertHttpBadRequest(response)
 
-        # Invalid email address
-        response = self.client.post(self.url, {
-            "email": "invalid@example.com",
+    def test_email_in_set_session(self):
+        # Login and check email in session
+        data = {
+            "email": self.EMAIL,
             "password": self.PASSWORD,
-        })
-        self.assertHttpBadRequest(response)
+        }
 
-    @ddt.data(True, False)
-    def test_missing_login_params(self, is_api_v1):
-        email_field_name = "email" if is_api_v1 else "email_or_username"
-        url = self.url if is_api_v1 else self.url_v2
-        # Missing password
-        response = self.client.post(url, {
-            email_field_name: self.EMAIL,
-        })
-        self.assertHttpBadRequest(response)
+        response = self.client.post(self.url, data)
+        self.assertHttpOK(response)
 
-        # Missing email
-        response = self.client.post(url, {
-            "password": self.PASSWORD,
-        })
-        self.assertHttpBadRequest(response)
+        # Ensure the login was successful
+        self.assertEqual(response.status_code, 200)
 
-        # Missing both email and password
-        response = self.client.post(url, {})
-        self.assertHttpBadRequest(response)
+        stored_email = self.client.session.get('email')
+        assert stored_email == self.EMAIL
```

Comment:
1. This is the important part of this test, but it is out of context of the middleware. I wonder if this test could be removed in favor of the integration test I described in the middleware, which might call this endpoint to update the email?
2. Do we need to separately call/test with the registration endpoint?

Reply:
> Do we need to separately call/test with the registration endpoint?

Thanks. This was an important part we were missing. We were not setting user email in session on registration.

Reply:
See other comment about potential alternative solution. Either way, I am marking this resolved. Thanks.